src/checks/stylelint.ts

```diff
@@ -0,0 +1,65 @@
+import { execaCommand } from 'execa';
+import { LintResult } from 'stylelint';
+import { projectPackageJson } from '../util/packageJson';
+
+const projectDependencies = projectPackageJson.dependencies || {};
+
+export const supportedFileExtensions = [
+  'css',
+  'sass',
+  'scss',
+  'less',
+  'js',
+  'tsx',
+  'jsx',
+];
+
+export const title = 'Stylelint';
+
+export default async function stylelintCheck() {
+  // Continue only if project is Next.js or UpLeveled React App
+  if (
+    !(
+      '@upleveled/react-scripts' in projectDependencies ||
+      'next' in projectDependencies
+    )
+  ) {
+    return;
+  }
+
+  try {
+    await execaCommand(
+      `yarn stylelint **/*.{${supportedFileExtensions.join(
+        ',',
+      )}} --max-warnings 0 --formatter json`,
+    );
+  } catch (error) {
+    const { stdout } = error as { stdout: string };
+
+    const stylelintJSONOutput = stdout.slice(
+      stdout.indexOf('['),
+      stdout.lastIndexOf(']') + 1,
+    );
+
+    // If no Stylelint errors detected, throw the error
+    if (!/"errored":/.test(stylelintJSONOutput)) {
```

Comment:
hm, this `"errored"` is not being typechecked here, maybe we can make this a variable that you also use in the `.filter` thing below to make sure that whatever property you're using is being correctly used in both places

Reply:
42eff47